src/fios_gui.cpp

```diff
@@ -666,8 +678,15 @@ struct SaveLoadWindow : public Window {
 			}
 		} else if (this->IsWidgetLowered(WID_SL_SAVE_GAME)) { // Save button clicked
 			if (this->abstract_filetype == FT_SAVEGAME || this->abstract_filetype == FT_SCENARIO) {
-				_switch_mode = SM_SAVE_GAME;
-				FiosMakeSavegameName(_file_to_saveload.name, this->filename_editbox.text.buf, lastof(_file_to_saveload.name));
+				if (_load_check_data.settings.game_creation.generation_unique_id != 0 && /* Don't warn if the save has no id (old save) */
+		_load_check_data.settings.game_creation.generation_unique_id != _settings_game.game_creation.generation_unique_id) {
```

Comment:
Indentation on this line is still bad. It should be two tabs past the "if" line, so six tabs in total. Matching what you do on line 411-413.